stylize.py

```diff
@@ -96,7 +96,12 @@ def stylize(network, initial, content, styles, iterations,
         loss = content_loss + style_loss + tv_loss
 
         # optimizer setup
-        train_step = tf.train.AdamOptimizer(learning_rate).minimize(loss)
+
+        def train_step(session = None ,iterations = 0):
+            if optimizer == 'adam':
+                tf.train.AdamOptimizer(learning_rate).minimize(loss)
+            elif optimizer == 'lbfgs':
+                tf.contrib.opt.ScipyOptimizerInterface(loss, method='L-BFGS-B', options={'maxiter': iterations}).minimize(session)
```

Comment:
I haven't tried this specific thing, but this seems kind of odd.

Also, in general – LBFGS is a stateful optimization method. So doing 1 iteration of LBFGS N times vs doing N iterations of LBFGS once wouldn't produce the same results (because the former would essentially throw away the state every time). So we need a good way of doing `iterations` iterations while being able to probe at progress (print stuff / generate checkpoint images).